File: Java/3 - Starbuzz/src/main/java/com/mycompany/starbuzz/Soy.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mycompany.starbuzz;

/**
 *
 * @author ismas
 *
 */
public class Soy extends CondimentDecorator {
    public Soy( Beverage beverage ){
        this.beverage = beverage;
    }
    
    @Override
    public String getDescription(){
        return this.beverage.getDescription() + ", Soy";
    }
    
    @Override
    public double cost(){
        switch(this.getSize()){
            case TALL:
                return this.beverage.cost() + 0.10;
            case GRANDE:
                return this.beverage.cost() + 0.15;
            case VENTI:
                return this.beverage.cost() + 0.20;
            default:
                return this.beverage.cost() + 0.10;
        }
    }
}

File: C#/3. Starbuzz (Decorator)/Classes/Soy.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Starbuzz.Classes
{
    public class Soy : CondimentDecorator
    {
        public Soy(Beverage beverage): base(beverage) { }

        public override string GetDescription()
        {
            return _Beverage.GetDescription() + ", Soy";
        }

        public override double Cost()
        {
            switch (_Size)
            {
                case Size.TALL:
                    return _Beverage.Cost() + 0.10;
                case Size.GRANDE:
                    return _Beverage.Cost() + 0.15;
                case Size.VENTI:
                    return _Beverage.Cost() + 0.20;
                default:
                    return _Beverage.Cost() + 0.10;
            }
        }
    }
}
